underscore.js

```diff
@@ -520,19 +520,35 @@
     if (iteratee != null) iteratee = cb(iteratee, context);
     var result = [];
     var seen = [];
+    var seenPrimitive = {};
     for (var i = 0, length = array.length; i < length; i++) {
       var value = array[i],
           computed = iteratee ? iteratee(value, i, array) : value;
+
       if (isSorted) {
         if (!i || seen !== computed) result.push(value);
         seen = computed;
-      } else if (iteratee) {
-        if (!_.contains(seen, computed)) {
-          seen.push(computed);
+      } else {
+        var vType = typeof computed,
+          isPrimitive = _.contains(['number', 'string', 'boolean', 'undefined'], vType);
```

Comment:
@akre54 I have implemented this, thanks for advice!
@michaelficarra I think I can patch isObject function to recognize symbols as primitives. Is it ok for a symbol to be a primitive?